### PR TITLE
use shared auditing from metal-lib

### PR DESCRIPTION
**Pull request by vknabel**
Based upon https://github.com/metal-stack/metal-lib/pull/73

Removes the internal auditing package in favor of the new package from metal-lib.
Implements https://github.com/metal-stack-cloud/docs/issues/95 for metal-stack.

Part of a group of PRs:

- https://github.com/metal-stack/metal-lib/pull/73
- https://github.com/metal-stack/metal-api/pull/425
- https://github.com/metal-stack-cloud/api-server/pull/226

### CODE REVIEW

**Comment by vknabel** [comment]
@Gerrit91 do you have an idea why the integration tests failed?
https://github.com/metal-stack/metal-api/actions/runs/4354209132/jobs/7609232493#step:4:211

From my perspective this should be unrelated.

File: go.mod [hunk 1]
```diff
@@ -9,7 +9,7 @@ require (
 	github.com/dustin/go-humanize v1.0.1
 	github.com/emicklei/go-restful-openapi/v2 v2.9.1
 	// FIXME go-restful v3.10.x breaks imageCreate
-	github.com/emicklei/go-restful/v3 v3.9.0
+	github.com/emicklei/go-restful/v3 v3.10.1
```
Comment (majst01) [inline go.mod:12]: This must not be done this breaks Integration tests

Comment (vknabel) [reply go.mod]: Thanks! Downgraded!

File: Makefile [hunk 2]
```diff
@@ -31,9 +31,18 @@ protoc-docker:
 	docker pull bufbuild/buf:1.14.0
 	docker run --rm --user $$(id -u):$$(id -g) -v $(PWD):/work --tmpfs /.cache -w /work/proto bufbuild/buf:1.14.0 generate -v
 
+.PHONY: metal-api
+metal-api:
```
Comment (Gerrit91) [inline Makefile:35]: What is this target used for? Isn't the mini-lab-push target sufficient?

File: go.mod [hunk 3]
```diff
@@ -19,7 +19,7 @@ require (
 	github.com/looplab/fsm v0.3.0
 	github.com/metal-stack/go-ipam v1.8.5
 	github.com/metal-stack/masterdata-api v0.9.0
-	github.com/metal-stack/metal-lib v0.11.4
+	github.com/metal-stack/metal-lib v0.11.5-0.20230308090023-340574de5f20
```
Comment (Gerrit91) [inline go.mod:22]: Can be pinned to latest release version.

File: Makefile (hunk 2 above)
Comment (vknabel) [reply Makefile:35]: Seems like a leftover from a different try. Thanks!

File: go.mod (hunk 3 above)
Comment (vknabel) [reply go.mod]: Pinned!

**Review by majst01 — APPROVED** [review]

File: go.mod [hunk 4]
```diff
@@ -189,3 +189,5 @@ require (
 	nhooyr.io/websocket v1.8.7 // indirect
 	tailscale.com v1.34.0 // indirect
 )
+
+require github.com/bufbuild/connect-go v1.5.2 // indirect
```
Comment (majst01) [inline go.mod:193]: Not as a seperate require please

**Review by majst01 — CHANGES_REQUESTED** [review]

File: go.mod (hunk 4 above)
Comment (vknabel) [reply go.mod]: Good catch! Haven't seen the separate require which was added by go mod tidy!

**Review by majst01 — APPROVED** [review]

**Pull request merged**